cmd/codegen/generator/functions.go

```diff
@@ -174,19 +175,19 @@ func (c *CommonFunctions) ConvertID(f introspection.Field) bool {
 // FormatInputType formats a GraphQL type into the SDK language input
 //
 // Example: `String` -> `string`
-func (c *CommonFunctions) FormatInputType(r *introspection.TypeRef) (string, error) {
-	return c.formatType(r, true)
+func (c *CommonFunctions) FormatInputType(r *introspection.TypeRef, scopes ...string) (string, error) {
+	return c.formatType(r, strings.Join(scopes, ""), true)
 }
 
 // FormatOutputType formats a GraphQL type into the SDK language output
 //
 // Example: `String` -> `string`
-func (c *CommonFunctions) FormatOutputType(r *introspection.TypeRef) (string, error) {
-	return c.formatType(r, false)
+func (c *CommonFunctions) FormatOutputType(r *introspection.TypeRef, scopes ...string) (string, error) {
+	return c.formatType(r, strings.Join(scopes, ""), false)
 }
 
 // formatType loops through the type reference to transform it into its SDK language.
-func (c *CommonFunctions) formatType(r *introspection.TypeRef, input bool) (representation string, err error) {
+func (c *CommonFunctions) formatType(r *introspection.TypeRef, scope string, input bool) (representation string, err error) {
```

Comment:
Note to self: this probably isn't the right approach for this.

We probably need to propagate the `scope` arguments down into the `formatTypeFuncs` interface below, so that we can let each codegen type decide how to apply the scope correctly.

Reply:
Resolved in [`c120c6f` (#6326)](https://github.com/dagger/dagger/pull/6326/commits/c120c6f3a6ee5c65d93ec0ece4036738fd270fc3).